### PR TITLE
Feature/inline text box

### PR DESCRIPTION
**Pull request by DanSallau**
inline-text

### CODE REVIEW

File: develop/components/pages/TablePage.tsx [hunk 1]
```diff
@@ -206,6 +216,17 @@ const TablePage: React.FunctionComponent = () => {
                         <Table columns={columns} data={smallData} onRowSelected={(rows: Array<TableRow>) => {}} />
                     </div>
 
+                    <p>Here is an example with inline edit</p>
+                    <div className="result wide">
+                        <div className="row">
+                            <div className="col text-right">
+                                <Button title="Cancel" label="Cancel" disabled={!editMode} onClick={() => setEditMode("cancel")} className="mr-2" />
+                                <Button title="Update" label={editMode === "edit" ? "Save" : "Edit"} onClick={() => setEditMode(editMode === "edit" ? "save" : "edit")} />
```
Comment (omarVodiaK) [inline develop/components/pages/TablePage.tsx]: label={editMode === "edit" ? "Save" : "Edit"} 
```suggestion
                                <Button title="Update" label={editMode === "edit" ? "Edit" : "Save"} onClick={() => setEditMode(editMode === "edit" ? "save" : "edit")} />
```

File: src/Table/Table.tsx [hunk 2]
```diff
@@ -932,7 +1032,8 @@ export const Table: React.FunctionComponent<TableProps> = React.memo(
                     selected: row.selected || false,
                     actionsDropdownDropped: row.actionsDropdownDropped || false,
                     expanded: row.expanded || false,
-                    subRows: row.subRows ? getRows(row.subRows) : []
+                    subRows: row.subRows ? getRows(row.subRows) : [],
+                    isEditMode: row.isEditMode || false
```
Comment (kherP) [inline src/Table/Table.tsx]: why you need `|| false` here

Comment (DanSallau) [reply src/Table/Table.tsx]: Its initialisation, isEditMode is optional and just like the case is for the remaining objects in the data, we initialized all in getRows .

File: develop/components/pages/TablePage.tsx (hunk 1 above)
Comment (DanSallau) [reply develop/components/pages/TablePage.tsx]: No, the current one is correct omar. Edit comes before save, the default button text is edit

File: src/Table/Table.tsx (hunk 2 above)
Comment (mario-subo) [reply src/Table/Table.tsx]: `!!row.expanded` <-- I prefer

**Pull request merged**